### PR TITLE
AM-468: Test setup for SmokeTests for ccd-case-document-api service

### PR DESCRIPTION
**Pull request by SivaKanukolluQA**
**Before creating a pull request make sure that:**
Test setup for SmokeTests for ccd-case-document-api service

### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/AM-468

### Change description ###
[ ] Yes
[x] No

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -92,7 +99,6 @@ sourceSets {
       runtimeClasspath += main.output
       srcDir file('src/smokeTest/java')
     }
-    resources.srcDir file('src/smokeTest/resources')
```
Comment (aashishaggarwal123) [inline build.gradle]: Don't we need resource directory for the smoke testing?

File: build.gradle [hunk 2]
```diff
@@ -128,7 +143,8 @@ checkstyle {
 
 pmd {
   toolVersion = "6.18.0"
-  sourceSets = [sourceSets.main, sourceSets.test, sourceSets.functionalTest, sourceSets.integrationTest, sourceSets.smokeTest]
+  //sourceSets = [sourceSets.main, sourceSets.test, sourceSets.functionalTest, sourceSets.integrationTest, sourceSets.smokeTest]
+  sourceSets = [sourceSets.test]
```
Comment (aashishaggarwal123) [inline build.gradle]: I think, this is required for all the sources. Could you please put some comments here highlighting it needs to be enabled for all the sources later.

File: build.gradle [hunk 3]
```diff
@@ -254,17 +270,41 @@ dependencies {
...
+  testCompile group: 'io.rest-assured', name: 'rest-assured', version: '3.1.0'
+  compile group: 'io.cucumber', name: 'cucumber-java', version: '5.0.0'
+  testCompile group: 'io.cucumber', name: 'cucumber-junit', version: '5.0.0'
+  testCompile 'io.github.openfeign:feign-jackson:9.7.0'
+
+  smokeTestCompile group: 'net.serenity-bdd', name: 'serenity-core', version: versions.serenity
+  smokeTestCompile group: 'net.serenity-bdd', name: 'serenity-junit', version: versions.serenity
+  smokeTestCompile group: 'net.serenity-bdd', name: 'serenity-rest-assured', version: versions.serenity
+  smokeTestCompile group: 'net.serenity-bdd', name: 'serenity-spring', version: versions.serenity
+  smokeTestCompile group: 'io.rest-assured', name: 'rest-assured', version: '3.3.0'
```
Comment (aashishaggarwal123) [inline build.gradle]: Can you please replace the version numbers with variables wherever possible.

File: src/functionalTest/java/uk/gov/hmcts/reform/ccd/document/am/befta/DataStoreBeftaMain.java [hunk 4]
```diff
@@ -0,0 +1,12 @@
+package uk.gov.hmcts.reform.ccd.document.am.befta;
```
Comment (aashishaggarwal123) [inline src/functionalTest/java/uk/gov/hmcts/reform/ccd/document/am/befta/DataStoreBeftaMain.java]: application name is ccd-case-document-am, so package name should also be same.

File: src/functionalTest/java/uk/gov/hmcts/reform/ccd/document/am/befta/DataStoreTestAutomationAdapter.java [hunk 5]
```diff
@@ -0,0 +1,126 @@
+package uk.gov.hmcts.reform.ccd.document.am.befta;
```
Comment (aashishaggarwal123) [inline src/functionalTest/java/uk/gov/hmcts/reform/ccd/document/am/befta/DataStoreTestAutomationAdapter.java]: Package naming needs adjustment.

File: src/functionalTest/java/uk/gov/hmcts/reform/ccd/document/am/befta/DataStoreTestAutomationAdapter.java [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+import uk.gov.hmcts.befta.BeftaMain;
+import uk.gov.hmcts.befta.DefaultTestAutomationAdapter;
+import uk.gov.hmcts.befta.data.UserData;
+import uk.gov.hmcts.befta.exception.FunctionalTestException;
+
+import java.io.File;
+import java.util.HashMap;
+import java.util.Map;
+
+public class DataStoreTestAutomationAdapter extends DefaultTestAutomationAdapter {
```
Comment (aashishaggarwal123) [inline src/functionalTest/java/uk/gov/hmcts/reform/ccd/document/am/befta/DataStoreTestAutomationAdapter.java]: class name needs adjustments as per this application.

File: src/functionalTest/java/uk/gov/hmcts/reform/ccd/document/am/befta/DataStoreTestAutomationAdapter.java [hunk 7]
```diff
@@ -0,0 +1,126 @@
...
+
+    private Logger logger = LoggerFactory.getLogger(DataStoreTestAutomationAdapter.class);
+
+    private static final String[] TEST_DEFINITIONS_NEEDED_FOR_TA = {
+        "src/functionalTest/resources/CCD_CNP_27.xlsx",
+        "src/functionalTest/resources/CCD_CNP_27_AUTOTEST1.xlsx",
+        "src/functionalTest/resources/CCD_CNP_27_AUTOTEST2.xlsx",
+        "src/functionalTest/resources/CCD_CNP_RDM5118.xlsx",
+
+        "src/functionalTest/resources/CCD_BEFTA_JURISDICTION1.xlsx",
```
Comment (aashishaggarwal123) [inline src/functionalTest/java/uk/gov/hmcts/reform/ccd/document/am/befta/DataStoreTestAutomationAdapter.java]: Please remove loading of all excel files except CCD_BEFTA_JURISDICTION2. Kindly delete corresponding files as well.

File: src/functionalTest/java/uk/gov/hmcts/reform/ccd/document/am/befta/DataStoreTestAutomationAdapter.java [hunk 8]
```diff
@@ -0,0 +1,126 @@
...
+
+            { "caseworker-befta_jurisdiction_2", "PUBLIC" },
+            { "caseworker-befta_jurisdiction_2-solicitor_1", "PUBLIC" },
+            { "caseworker-befta_jurisdiction_2-solicitor_2", "PUBLIC" },
+            { "caseworker-befta_jurisdiction_2-solicitor_3", "PUBLIC" },
+            { "citizen", "PUBLIC" },
+
+            { "caseworker-befta_jurisdiction_3", "PUBLIC" },
+            { "caseworker-befta_jurisdiction_3-solicitor", "PUBLIC" }
+    };
```
Comment (aashishaggarwal123) [inline src/functionalTest/java/uk/gov/hmcts/reform/ccd/document/am/befta/DataStoreTestAutomationAdapter.java]: Kindly keep only CCD_BEFTA_JURISDICTION2 specific roles for now.

File: src/functionalTest/java/uk/gov/hmcts/reform/ccd/document/am/befta/DataStoreTestDataLoader.java [hunk 9]
```diff
@@ -0,0 +1,10 @@
+package uk.gov.hmcts.reform.ccd.document.am.befta;
+
+@SuppressWarnings("checkstyle:HideUtilityClassConstructor")
+public class DataStoreTestDataLoader {
```
Comment (aashishaggarwal123) [inline src/functionalTest/java/uk/gov/hmcts/reform/ccd/document/am/befta/DataStoreTestDataLoader.java]: naming needs adjustments

File: src/smokeTest/java/uk/gov/hmcts/reform/ccd/document/am/SmokeTest.java [hunk 10]
```diff
@@ -0,0 +1,36 @@
+package uk.gov.hmcts.reform.ccd.document.am;
```
Comment (aashishaggarwal123) [inline src/smokeTest/java/uk/gov/hmcts/reform/ccd/document/am/SmokeTest.java]: package  name needs adjustment.

File: build.gradle (hunk 1 above)
Comment (SivaKanukolluQA) [reply build.gradle]: As of now not required. We can add when ever it is necessary.

File: build.gradle (hunk 2 above)
Comment (SivaKanukolluQA) [reply build.gradle]: Updated comments.

File: build.gradle (hunk 3 above)
Comment (SivaKanukolluQA) [reply build.gradle]: Updated:
  compile group: 'io.cucumber', name: 'cucumber-java', version: versions.cucumber
  testCompile group: 'io.rest-assured', name: 'rest-assured', version: versions.rest_assured
  testCompile group: 'io.cucumber', name: 'cucumber-junit', version: versions.cucumber
  testCompile group: 'io.github.openfeign', name: 'feign-jackson', version: versions.feign_jackson

  smokeTestCompile group: 'net.serenity-bdd', name: 'serenity-core', version: versions.serenity
  smokeTestCompile group: 'net.serenity-bdd', name: 'serenity-junit', version: versions.serenity
  smokeTestCompile group: 'net.serenity-bdd', name: 'serenity-rest-assured', version: versions.serenity
  smokeTestCompile group: 'net.serenity-bdd', name: 'serenity-spring', version: versions.serenity
  smokeTestCompile group: 'io.rest-assured', name: 'rest-assured', version: versions.rest_assured

File: src/functionalTest/java/uk/gov/hmcts/reform/ccd/document/am/befta/DataStoreTestAutomationAdapter.java (hunk 8 above)
Comment (SivaKanukolluQA) [reply src/functionalTest/java/uk/gov/hmcts/reform/ccd/document/am/befta/DataStoreTestAutomationAdapter.java]: Removed other data entries except Jurisdiction2.

File: src/functionalTest/java/uk/gov/hmcts/reform/ccd/document/am/befta/DataStoreTestAutomationAdapter.java (hunk 7 above)
Comment (SivaKanukolluQA) [reply src/functionalTest/java/uk/gov/hmcts/reform/ccd/document/am/befta/DataStoreTestAutomationAdapter.java]: Removed other excel files.

File: src/functionalTest/java/uk/gov/hmcts/reform/ccd/document/am/befta/DataStoreTestAutomationAdapter.java (hunk 6 above)
Comment (SivaKanukolluQA) [reply src/functionalTest/java/uk/gov/hmcts/reform/ccd/document/am/befta/DataStoreTestAutomationAdapter.java]: Updated.  CaseDocumentAmTestAutomationAdapter

File: src/functionalTest/java/uk/gov/hmcts/reform/ccd/document/am/befta/DataStoreTestDataLoader.java (hunk 9 above)
Comment (SivaKanukolluQA) [reply src/functionalTest/java/uk/gov/hmcts/reform/ccd/document/am/befta/DataStoreTestDataLoader.java]: Updated. CaseDocumentAmTestDataLoader

File: src/smokeTest/java/uk/gov/hmcts/reform/ccd/document/am/SmokeTest.java (hunk 10 above)
Comment (SivaKanukolluQA) [reply src/smokeTest/java/uk/gov/hmcts/reform/ccd/document/am/SmokeTest.java]: Build is failing. Not able to update this comment at the moment.
Will update this change later.

File: src/functionalTest/java/uk/gov/hmcts/reform/ccd/document/am/befta/DataStoreTestAutomationAdapter.java (hunk 5 above)
Comment (SivaKanukolluQA) [reply src/functionalTest/java/uk/gov/hmcts/reform/ccd/document/am/befta/DataStoreTestAutomationAdapter.java]: Build is failing. Not able to update this comment at the moment.
Will update this change later. 
Execution failed for task ':checkOutcomes'.
> No such property: toUri for class: sun.nio.fs.UnixPath

**Pull request merged**